testing/internal/e2e/tests/base/target_tcp_connect_session_max_seconds_test.go

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package base_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/session"
+	"github.com/hashicorp/boundary/internal/target"
+	"github.com/hashicorp/boundary/testing/internal/e2e"
+	"github.com/hashicorp/boundary/testing/internal/e2e/boundary"
+	"github.com/stretchr/testify/require"
+)
+
+// TestCliTcpTargetConnectTargetWithSessionMaxSeconds connects to a target that
+// has session max seconds defined. An error should return when trying to
+// connect to the target after the time limit has been reached.
+func TestCliTcpTargetConnectTargetWithSessionMaxSeconds(t *testing.T) {
```

Comment:
I did have one test variant where...
- started `boundary connect` in a goroutine
- kept creating new channels (using `ssh localhost`) until we reach the time limit
- confirm that trying to create a new channel after the time limit results in an error

Is it worth also having this other variant as a test or this one sufficient?

Reply:
Probably worth having a test to see that new connections are rejected, in addition to existing ones being torn down, yeah.

Reply:
Added another test